server/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -577,6 +564,21 @@ public static SortField readSortField(StreamInput in) throws IOException {
         return sortField;
     }
 
+    public static SortField[] readSortFields(StreamInput in) throws IOException {
+        SortField[] fields = new SortField[in.readVInt()];
```

Comment:
can't you use writeArray with a writer? I think that would be good instead of iterating manually. same on teh read end

Reply:
yes, not sure why I missed this ;P